controllers/configuration_controller.go

```diff
@@ -1336,25 +1483,42 @@ func (meta *TFConfigurationMeta) RenderConfiguration(configuration *v1beta2.Conf
 	}
 }
 
-// GetGitCredentialsSecret will get the secret containing the SSH private key & known_hosts
-func GetGitCredentialsSecret(ctx context.Context, k8sClient client.Client, secretRef *v1.SecretReference) (*v1.Secret, error) {
+func GetSecretOrConfigMap(ctx context.Context, k8sClient client.Client, isSecret bool, ref *v1.SecretReference, neededKeys []string, errMsg string, keyErrMsg string) (metav1.Object, error) {
```

Comment:
`errMsg` and `keyErrMsg` is duplicated. You can pass a parameter named `errKey` like "git credential" and compose the error message in `GetSecretOrConfigMap`.

```go
errors.Errorf("%s not found in %s %s", key, errKey, "sercret" or "configMap")
```
The last arg can be determined by `isSecret`

Reply:
fixed.